6_visualize/src/prep_spark_line_fun.R

```diff
@@ -1,63 +1,184 @@
+# This function prepares the coordinates and shapes (polygons and line) for each
+# site as much as is possible without yet knowing the final plot view
+# coordinates. This function is called from both prep_spark_lines_fun and
+# prep_spark_starts_fun.
+prep_spark_funs_data <- function(storm_data, site_data, timestep_ind, spark_config, DateTime) {
+  # Compute the full x limits for all datetimes
+  storm_timesteps <- fetch_read(timestep_ind)
+  date_lims <- as.POSIXct(range(storm_timesteps), tz = "UTC")
 
-prep_spark_line_fun <- function(storm_data, date_config, spark_config, gage_col_config, DateTime){
-  sites <- unique(storm_data$site_no)
+  # Choose order of sparklines
+  sites <- site_data %>%
+    bind_cols(., as_data_frame(sf::st_coordinates(.))) %>%
+    sf::st_set_geometry(NULL) %>%
```

Comment:
True...I think I put it in there in case we wanted to do any more data-framey things with the site_data to come up with an appealing ordering of sites. Since it doesn't actually hurt anything, and might eventually help, I'd like to keep it in.